zulipterminal/ui_tools/boxes.py

```diff
@@ -1676,10 +1676,20 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
             self.model.controller.view.write_box.msg_write_box.set_edit_pos(len(quote))
             self.model.controller.view.middle_column.set_focus("footer")
         elif is_command_key("EDIT_MESSAGE", key):
-            if self.message["sender_id"] != self.model.user_id:
-                self.model.controller.report_error(
-                    " You can't edit messages sent by other users."
-                )
+            # User can't edit messages of others that already have a topic
+            if (
+                self.message["sender_id"] != self.model.user_id
+                and self.message["subject"] != "(no topic)"
```

Comment:
We mentioned this in the test, but might be worth noting here that a private subject is ""?
That said, can you move the subject check into the stream conditional? That seems cleaner.

Reply:
I think we can remove this whole block from here?
They should already be handled fine with the below conditionals.

Reply:
We can't remove this block. It handles the "anytime" message-edit case - almost forgot about that earlier. :sweat_smile: 

I don't think we can move the subject check into the stream conditional below too. We don't enter the next block since `realm_message_content_edit_limit_seconds` will be `0` for "anytime" message-edit, so we need to handle that case outside either in the beginning or end?